pymc/initial_point.py

```diff
@@ -93,21 +107,18 @@ def make_initial_point_fns_per_chain(
                 return_transformed=True,
             )
         ] * chains
-    elif len(overrides) == chains:
-        ipfns = [
-            make_initial_point_fn(
-                model=model,
-                jitter_rvs=jitter_rvs,
-                overrides=chain_overrides,
-                return_transformed=True,
-            )
-            for chain_overrides in overrides
-        ]
-    else:
-        raise ValueError(
-            f"Number of initval dicts ({len(overrides)}) does not match the number of chains ({chains})."
+        return ipfns
+
+    assert isinstance(overrides, Sequence) and len(overrides) == chains
```

Comment:
Already checked above

Reply:
It's also related to the meaning of `assert`. I see it less as a control flow, more as an executable comment. Converting this to a comment would leave the possibility that the comment goes out of date.
Is this function typically hot?
Also, asserts are disabled by `python -O`, perhaps this is something that we could do?

Reply:
I don't think anyone ever disables asserts tbh